numpy/lib/tests/test_function_base.py

```diff
@@ -658,6 +658,17 @@ def test_nd(self):
         assert_array_equal(diff(x, axis=0), out3)
         assert_array_equal(diff(x, n=2, axis=0), out4)
 
+    def test_n(self):
+        x = list(range(3))
+        assert_raises(ValueError, diff, x, n=-1)
+        output = [diff(x, n=n) for n in range(1, 5)]
+        expected = [[1, 1], [0], [], []]
+        assert_(diff(x, n=0) is x)
```

Comment:
The previous behavior is to return the object itself. I documented that and added a test, but I want to avoid making an API change. Instead of a copy, I would probably go with at least an application of `asanyarray`.